packages/radix-ui-themes/src/components/checkbox-card-group.tsx

```diff
@@ -0,0 +1,122 @@
+  CHECKBOX_CARD_GROUP_NAME,
+  [createCheckboxGroupScope]
+);
+const useCheckboxGroupScope = createCheckboxGroupScope();
+
+type CheckboxCardGroupContextValue = {
+  size?: Responsive<(typeof checkboxCardGroupPropDefs.size.values)[number]>;
+};
+
+const [CheckboxCardGroupProvider, useCheckboxCardGroupContext] =
+  createCheckboxCardGroupContext<CheckboxCardGroupContextValue>(CHECKBOX_CARD_GROUP_NAME);
+
+type CheckboxCardGroupRootElement = React.ElementRef<typeof CheckboxGroupPrimitive.Root>;
+type CheckboxCardGroupRootOwnProps = GetPropDefTypes<typeof checkboxCardGroupPropDefs>;
+interface CheckboxCardGroupRootProps
+  extends PropsWithoutRefOrColor<typeof CheckboxGroupPrimitive.Root>,
+    MarginProps,
+    CheckboxCardGroupRootOwnProps {}
+const CheckboxCardGroupRoot = React.forwardRef<
+  CheckboxCardGroupRootElement,
+  CheckboxCardGroupRootProps
+>((props: ScopedProps<CheckboxCardGroupRootProps>, forwardedRef) => {
+  const { __scopeCheckboxCardGroup, className, color, ...rootProps } = extractProps(
+    props,
+    checkboxCardGroupPropDefs,
+    marginPropDefs
+  );
+  const checkboxGroupScope = useCheckboxGroupScope(__scopeCheckboxCardGroup);
+  return (
+    <CheckboxCardGroupProvider scope={__scopeCheckboxCardGroup} size={props.size}>
+      <Grid asChild>
+        <CheckboxGroupPrimitive.Root
+          {...checkboxGroupScope}
+          data-accent-color={color}
+          {...rootProps}
+          ref={forwardedRef}
+          className={classNames('rt-CheckboxCardGroupRoot', className)}
+        />
+      </Grid>
+    </CheckboxCardGroupProvider>
+  );
+});
+CheckboxCardGroupRoot.displayName = 'CheckboxCardGroupRoot';
+
+type CheckboxCardGroupItemElement = React.ElementRef<typeof CheckboxGroupPrimitive.Item>;
+interface CheckboxCardGroupItemProps
+  extends Omit<React.ComponentPropsWithoutRef<typeof CheckboxGroupPrimitive.Item>, 'asChild'>,
+    MarginProps {}
+const CheckboxCardGroupItem = React.forwardRef<
+  CheckboxCardGroupItemElement,
+  ScopedProps<CheckboxCardGroupItemProps>
+>(({ __scopeCheckboxCardGroup, children, className, ...props }, forwardedRef) => {
+  const context = useCheckboxCardGroupContext('item', __scopeCheckboxCardGroup);
+  const checkboxGroupScope = useCheckboxGroupScope(__scopeCheckboxCardGroup);
+  const { className: checkboxClassName } = extractProps(
+    // Pass size value from the context and static variant to generate styles
+    { size: context?.size, variant: 'surface', className: '' },
+    // Pass size & variant prop defs to allow it to be extracted
+    baseCheckboxPropDefs
+  );
+  return (
+    <label className={classNames('rt-reset', 'rt-CheckboxCardGroupItem', className)}>
+      {children}
+      <CheckboxGroupPrimitive.Item
+        {...checkboxGroupScope}
+        {...props}
+        ref={forwardedRef}
+        className={classNames(
+          'rt-reset',
+          'rt-BaseCheckboxRoot',
+          checkboxClassName,
+          'rt-CheckboxCardGroupItemCheckbox'
+        )}
+      >
```

Comment:
Should we pass the high contrast class manually here?

Reply:
To render a black checkbox

Reply:
Oh nice catch, yeah they don't look right.